### ISSUE TITLE
Execute code quality checks and apply fixes for critical items

### ISSUE THREAD

**Issue by mattrpav** [atlasmap/atlasmap#59]

**Comment by rhuss** [comment]
I'm going to integrate codacy today with the checkstyle rules which we are using in Syndesis.

**Comment by mattrpav** [comment]
I ran the latest build of SonarQube against the atlasmap code base and have made a first pass on the housekeeping tasks.

Key notes:

1. the atlas-demo-deps accounted for 90% of initial issues (5,000+) and 27k of the total 41k lines of code. This module was mostly a copy of the org.apache.camel.salesforce.* object models used for the RH Summit demo. This module has now been removed from the code base as its outdated and unused.

2. Code base is now down to a much more reasonable 14k lines

3. Unit test coverage measurement is reporting ~50% on 800+ unit tests; however, that is surely incorrect, b/c SonarQube/JaCoCo don't account for the lines covered in the itests. For example, most of the *Module.java classes show 0% - 16% line coverage, despite the fact we have 180+ reference mapping tests under atlas-itests-reference-mappings.

4. 0 Vulnerabilities, 0 Blocker, 0 Critical bugs

5. 99 Reported bugs, a good chunk (30% or so) of those relate to operations related to DataConversion, many of the operations would be expected to throw off a report.
